### ISSUE TITLE
Runtime Warning: RNN module weights are not part of single contiguous chunk of memory.

### ISSUE THREAD

**Issue by usptact** [zihangdai/mos#11]
I am trying to train the model under Windows 10 with CUDA 9.2 (NVIDIA 1070 GPU).

Pytorch version 1.0.0

I get the following error when trying to build from wikitext-2:

```
C:\Users\vlad\Anaconda3\envs\py36\python.exe C:/Users/vlad/Documents/GitHub/mos/main.py --epochs 1000 --data data/wikitext-2 --save WT2 --dropouth 0.2 --seed 1882 --n_experts 15 --nhid 1150 --nhidlast 650 --emsize 300 --batch_size 15 --lr 15.0 --dropoutl 0.29 --small_batch_size 5 --max_seq_len_delta 20 --dropouti 0.55 --single_gpu
Experiment dir : WT2-20181223-005932
torch.Size([139241, 15])
torch.Size([21764, 10])
torch.Size([245569, 1])
Applying weight drop of 0.5 to weight_hh_l0
Applying weight drop of 0.5 to weight_hh_l0
Applying weight drop of 0.5 to weight_hh_l0
param size: 34909528
Args: Namespace(alpha=2, batch_size=15, beta=1, bptt=70, clip=0.25, continue_train=False, cuda=True, data='data/wikitext-2', dropout=0.4, dropoute=0.1, dropouth=0.2, dropouti=0.55, dropoutl=0.29, emsize=300, epochs=1000, log_interval=200, lr=15.0, max_seq_len_delta=20, model='LSTM', n_experts=15, nhid=1150, nhidlast=650, nlayers=3, nonmono=5, save='WT2-20181223-005932', seed=1882, single_gpu=True, small_batch_size=5, tied=True, wdecay=1.2e-06, wdrop=0.5)
Model total parameters: 34909528
C:\Users\vlad\Anaconda3\envs\py36\lib\site-packages\torch\nn\modules\rnn.py:179: RuntimeWarning: RNN module weights are not part of single contiguous chunk of memory. This means they need to be compacted at every call, possibly greatly increasing memory usage. To compact weights again call flatten_parameters().
  self.dropout, self.training, self.bidirectional, self.batch_first)
Traceback (most recent call last):
  File "C:/Users/vlad/Documents/GitHub/mos/main.py", line 261, in <module>
    train()
  File "C:/Users/vlad/Documents/GitHub/mos/main.py", line 205, in train
    log_prob, hidden[s_id], rnn_hs, dropped_rnn_hs = parallel_model(cur_data, hidden[s_id], return_h=True)
  File "C:\Users\vlad\Anaconda3\envs\py36\lib\site-packages\torch\nn\modules\module.py", line 489, in __call__
    result = self.forward(*input, **kwargs)
  File "C:\Users\vlad\Documents\GitHub\mos\model.py", line 84, in forward
    raw_output, new_h = rnn(raw_output, hidden[l])
  File "C:\Users\vlad\Anaconda3\envs\py36\lib\site-packages\torch\nn\modules\module.py", line 489, in __call__
    result = self.forward(*input, **kwargs)
  File "C:\Users\vlad\Documents\GitHub\mos\weight_drop.py", line 47, in forward
    return self.module.forward(*args)
  File "C:\Users\vlad\Anaconda3\envs\py36\lib\site-packages\torch\nn\modules\rnn.py", line 179, in forward
    self.dropout, self.training, self.bidirectional, self.batch_first)
RuntimeError: shape '[5290000, 1]' is invalid for input of size 4600
```

**Comment by ashu5644** [comment]
Hey, @usptact  I am also getting similar problem with ptb dataset. Did you get solution of problem? 

Traceback (most recent call last):                                                                                                                                                                                   File "main.py", line 270, in <module>                                                                                                                                                                                train()                                                                                                                                                                                                          File "main.py", line 211, in train                                                                                                                                                                                   log_prob, hidden[s_id], rnn_hs, dropped_rnn_hs = parallel_model(cur_data, hidden[s_id], return_h=True)                                                                                                           File "C:\Users\Anaconda3\envs\tensorflow\lib\site-packages\torch\nn\modules\module.py", line 489, in __call__                                                                                               result = self.forward(*input, **kwargs)                                                                                                                                                                          File "D:\mos-master\model.py", line 84, in forward                                                                                                                                                                   raw_output, new_h = rnn(raw_output, hidden[l])                                                                                                                                                                   File "C:\Users\Anaconda3\envs\tensorflow\lib\site-packages\torch\nn\modules\module.py", line 489, in __call__                                                                                               result = self.forward(*input, **kwargs)                                                                                                                                                                          File "D:\mos-master\weight_drop.py", line 47, in forward                                                                                                                                                             return self.module.forward(*args)                                                                                                                                                                                File "C:\Users\Anaconda3\envs\tensorflow\lib\site-packages\torch\nn\modules\rnn.py", line 179, in forward                                                                                                   self.dropout, self.training, self.bidirectional, self.batch_first)                                                                                                                                             RuntimeError: shape '[3686400, 1]' is invalid for input of size 3840

**Comment by usptact** [comment]
@ashu826 Unfortunately not.

**Comment by nkcr** [comment]
Have you tried using pytorch 0.4.1 ? Wrong pytorch version could be the reason.

**Comment by ashu5644** [comment]
yes It was problem of wrong version. I solved it using pytorch 0.2.0
